### PR TITLE
Update for Angular 15

### PR DESCRIPTION
**Pull request by coverbeck**
Update ng2-ui-auth for Angular 15.

This repo has builds committed to its `/dist` directory. We should probably automate it, but I suspect it will change infrequently and I'd like to get the migration to Angular 16 unblocked (not a typo -- first bringing this into Angular 15, to limit the scope, then we'll the UI upgrade to Angular 16 separately).

To review the non-build changes, look at the `package.json`, `package-lock.json`, `public_api.ts` and `ng2-ui-auth-module.ts`, and `README.md` files.

I have tested this locally with Dockstore, using the technique described in @david4096 's comment [here](https://github.com/dockstore/dockstore-ui2/pull/1984#pullrequestreview-2155014652) -- there will be a corresponding UI pr that uses this module directly from GitHub, without npm.

dockstore/dockstore#3641
DOCK-1475

### CODE REVIEW

File: projects/ng2-ui-auth/src/lib/ng2-ui-auth.module.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import { AuthService } from './auth.service';
   exports: []
 })
 export class Ng2UiAuthModule {
-  static forRoot(configOptions?: IPartialConfigOptions, defaultJwtInterceptor = true): ModuleWithProviders {
+  static forRoot(configOptions?: IPartialConfigOptions, defaultJwtInterceptor = true): ModuleWithProviders<Ng2UiAuthModule> {
```
Comment (coverbeck) [inline projects/ng2-ui-auth/src/lib/ng2-ui-auth.module.ts:21]: This change needed for newer versions of Angular.

File: projects/ng2-ui-auth/src/public_api.ts [hunk 2]
```diff
@@ -28,8 +29,9 @@ export {
   BrowserStorageService,
   AuthService,
   ConfigService,
+  IPartialConfigOptions,
```
Comment (coverbeck) [inline projects/ng2-ui-auth/src/public_api.ts:32]: This was not exposed, and we use it. Not sure how it worked before; maybe the released version didn't match master?

File: README.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+Forked from ronzeidman/ng2-ui-auth, which has not been updated for newer versions of Angular. We will
```
Comment (denis-yuen) [inline README.md:1]: "We will" seems cut off

**Review by denis-yuen — APPROVED** [review]

**Comment by coverbeck** [comment]
Not going to do this, see [this comment](https://github.com/dockstore/dockstore-ui2/pull/1984#issuecomment-2223528417).

**Pull request closed without merging**